publisher/server/publisher.go

```diff
@@ -22,9 +22,8 @@ const (
 type Server struct {
 	DockerClient *docker.Client
```

Comment:
Perhaps you should also switch this to an interface. So we can make a mock docker client in tests..